.github/workflows/package-manager-demo.yml

```diff
@@ -11,42 +11,51 @@ permissions:
 
 jobs:
   build:
-    env:
-      PACKAGEMANAGER_TOKEN_SOLO: ${{ secrets.PACKAGEMANAGER_TOKEN_SOLO }}
-      PACKAGEMANAGER_TOKEN_CLUSTER: ${{ secrets.PACKAGEMANAGER_TOKEN_CLUSTER }}
-      PACKAGEMANAGER_TOKEN_S3: ${{ secrets.PACKAGEMANAGER_TOKEN_S3 }}
     strategy:
       matrix:
         r-version: ['3.6.3', '4.1.3']
         os: [ {os: "ubuntu-20.04", distro: "focal"}, {os: "ubuntu-18.04", distro: "bionic"} ]
         package-manager: [
-          {token: "$PACKAGEMANAGER_TOKEN_SOLO", endpoint: "https://solo.rstudiopm.com"},
-          {token: "$PACKAGEMANAGER_TOKEN_S3", endpoint: "https://s3.rstudiopm.com"},
-          {token: "$PACKAGEMANAGER_TOKEN_CLUSTER", endpoint: "https://cluster.rstudiopm.com"},
+          {endpoint: "https://solo.rstudiopm.com", token: "PACKAGEMANAGER_TOKEN_SOLO"},
+          {endpoint: "https://s3.rstudiopm.com", token: "PACKAGEMANAGER_TOKEN_S3"},
+          {endpoint: "https://cluster.rstudiopm.com", token: "PACKAGEMANAGER_TOKEN_CLUSTER"},
         ]
     runs-on: ${{ matrix.os.os }}
     steps:
       - uses: actions/checkout@v3
       - name: Set up R ${{ matrix.r-version }}
-        uses: r-lib/actions/setup-r@f57f1301a053485946083d7a45022b278929a78a
+        uses: r-lib/actions/setup-r@v2
         with:
           r-version: ${{ matrix.r-version }}
       - name: Build source package
-        run: R CMD build .
+        run: |
+          source_dir=$(mktemp -d)
+          (cd $source_dir; R CMD build $GITHUB_WORKSPACE)
+          echo "SOURCE_FILE=$(readlink -f $source_dir/*)" >> $GITHUB_ENV
```

Comment:
Running `R CMD build|INSTALL` in a fresh temp directory is the best way I've seen to dynamically determine the built package file, but it's pretty ugly. There might be a better way to do this without having to use `$GITHUB_WORKSPACE` to find the original working dir, or `readlink` to get the full path to the built file.

An alternative method would be to use a higher-level R package to do this, which might be acceptable for workflows that already have R installed:
```r
> install.packages("pkgbuild")

# pkgbuild::build() returns the path to the built package file
> pkgbuild::build("~/package-manager-demo/")
✔  checking for file ‘/home/greg/package-manager-demo/DESCRIPTION’ ...
─  preparing ‘packageManagerDemo’:
✔  checking DESCRIPTION meta-information
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
   Omitted ‘LazyData’ from DESCRIPTION
─  building ‘packageManagerDemo_1.0.0.tar.gz’
   
[1] "/home/greg/packageManagerDemo_1.0.0.tar.gz"
```

Reply:
I replaced `$GITHUB_WORKSPACE` with bash's `$OLDPWD` to be easier to understand and use outside of GHA, and swapped `readlink` with an `ls -d` command.